### ISSUE TITLE
[Bug]: Storage, History, Subscriptions no longer available on current version. Import not working either!!!

### ISSUE THREAD

**Issue by vdbhb59** [MarmadileManteater/FreeTubeCordova#248]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

History, subs, saved data everything is lost upon upgrading to v0.19.1.109 Beta from F-Droid Izzy repo.
Moreover, if I try to import from PC app, it still does not get imported, no matter what. I have lost everything and unable to import.

### Expected Behavior

Data should not have been lost.
Moreover import should work too.

### Issue Labels

data loss

### FreeTube Version

v0.19.1.109 Beta

### Operating System Version

Android 13

### Installation Method

.apk

### Primary API used

Invidious API

### Last Known Working FreeTube Version (If Any)

0.18.0.107

### Additional Information

#152

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by MarmadileManteater** [comment]
This could potentially be a problem with some specific subscription/history entries in your files, but it is hard to tell without information.

 I recommend trying to subscribe to a few channels on a clean install and exporting the subscriptions and posting the file contents, and then, you should delete the subscriptions in app and reimport them to see what happens.

**Comment by vdbhb59** [comment]
> This could potentially be a problem with some specific subscription/history entries in your files, but it is hard to tell without information.
> 
> I recommend trying to subscribe to a few channels on a clean install and exporting the subscriptions and posting the file contents, and then, you should delete the subscriptions in app and reimport them to see what happens.

How? I had around 25+ subs, 100+ history items till I upgraded. The moment upgrade happened, everything vanished. I have not tried fresh install subs yet. Will redo steps once more.

**Comment by MarmadileManteater** [comment]
I have no idea why the local storage cleared on application upgrade, but it could be device specific (I can only test what I own). I understand your frustration with this issue, but the amount of information you are giving me is not enough to actually troubleshoot your issue.

**Comment by MarmadileManteater** [comment]
I just remembe

**Comment by MarmadileManteater** [comment]
Phone ui on GH is awful. didn't mean to close.

**Comment by MarmadileManteater** [comment]
I just remembered something that might be relevant. I made a change which allows clear text traffic (in order to allow connecting to locally hosted invidious servers similarly to the desktop app). That could have messed with local storage on app upgrade, but this does not explain why you can't import anything after upgrading.

**Comment by vdbhb59** [comment]
No worries. Not frustrated as such. Just confused. Thankfully I have same database on my PC and I have its backup.

I will redo clean install and see how it works out. I will revert in a day or 2. : )

**Comment by vdbhb59** [comment]
I tried with fresh install. New subs work, but import still does not work.
@MarmadileManteater

**Comment by MarmadileManteater** [comment]
Can you export the new subs? and if so, what file does it output? Are you still trying to import your full original subscriptions or are you trying to import the newly exported subscriptions?

**Comment by vdbhb59** [comment]
> Can you export the new subs? and if so, what file does it output? Are you still trying to import your full original subscriptions or are you trying to import the newly exported subscriptions?

Export from android app does nothing. Hence, I did a clean install on PC as well and added some subs, history etc and tried importing them onto android app. Does nothing.
However, as I said, subs, history etc is working as live on fresh android app install. Where can I get the logs, I cannot see it anywhere.

**Comment by OtBe21666** [comment]
Oddly enough, I can load my .db subs list on my 2019 Shield Pro but not my phone running Android 13?

**Comment by OtBe21666** [comment]
Oddly enough, I can load my .db subs list on my 2019 Shield Pro but not my phone running Android 13?

**Comment by vdbhb59** [comment]
Yup, same here @OtBe21666. I too am running A13. Probably A13 compatibility issue @MarmadileManteater ?
@OtBe21666 you have duplicate post above (probably network error). Kindly delete one. :)

**Comment by MarmadileManteater** [comment]
> > Can you export the new subs? and if so, what file does it output? Are you still trying to import your full original subscriptions or are you trying to import the newly exported subscriptions?
> 
> Export from android app does nothing. Hence, I did a clean install on PC as well and added some subs, history etc and tried importing them onto android app. Does nothing. However, as I said, subs, history etc is working as live on fresh android app install. Where can I get the logs, I cannot see it anywhere.

To get logs, you need to enable "USB debugging" on your device. Then, you will need to connect your phone to a computer and open chromium (or any chromium browser i think) and navigate to `chrome://inspect#devices`, and from there, it should let you inspect the webview that FreeTube Cordova is running in, and the logs should be in the console.

**Comment by vdbhb59** [comment]
> > > Can you export the new subs? and if so, what file does it output? Are you still trying to import your full original subscriptions or are you trying to import the newly exported subscriptions?
> > 
> > 
> > Export from android app does nothing. Hence, I did a clean install on PC as well and added some subs, history etc and tried importing them onto android app. Does nothing. However, as I said, subs, history etc is working as live on fresh android app install. Where can I get the logs, I cannot see it anywhere.
> 
> To get logs, you need to enable "USB debugging" on your device. Then, you will need to connect your phone to a computer and open chromium (or any chromium browser i think) and navigate to `chrome://inspect#devices`, and from there, it should let you inspect the webview that FreeTube Cordova is running in, and the logs should be in the console.

Okay. I will try that and give you the same.

**Comment by vdbhb59** [comment]
BTW, this is another issue, which hopefully should not get carried forward here..
https://github.com/FreeTubeApp/FreeTube/issues/4337

**Comment by vdbhb59** [comment]
> chrome://inspect#devices

Tried, but not sure I am getting the correct manner to do it, as I am not getting anything in the chrome inspect page other than "Discover" option and . I am using Ungoogled Chromium though.
```
Devices
 Discover USB devices Port forwarding...
 Discover network targets Configure...
Open dedicated DevTools for Node
```

**Comment by 321nick** [comment]
> > > Can you export the new subs? and if so, what file does it output? Are you still trying to import your full original subscriptions or are you trying to import the newly exported subscriptions?
> > 
> > 
> > Export from android app does nothing. Hence, I did a clean install on PC as well and added some subs, history etc and tried importing them onto android app. Does nothing. However, as I said, subs, history etc is working as live on fresh android app install. Where can I get the logs, I cannot see it anywhere.
> 
> To get logs, you need to enable "USB debugging" on your device. Then, you will need to connect your phone to a computer and open chromium (or any chromium browser i think) and navigate to `chrome://inspect#devices`, and from there, it should let you inspect the webview that FreeTube Cordova is running in, and the logs should be in the console.

Im on a Google Pixel, and i have a similar issue with importing of subscriptions not working. i tried USB debugging but i cant find where freetubeCordova comes up to check its logs. all i get is a list of devices, or if i have chrome open, i can see chromes webhistory on my own phone. but nothing with FreeTubeCordova. 

EDIT: i found the webview that freetube cordova isrunning in and i was watching the log as i tried importing subscriptions, using the .db file that i exported using the desktop app, but nothing comes up in it, and the import doesnt go through (no new subscriptions). 
![image](https://github.com/MarmadileManteater/FreeTubeCordova/assets/19196462/548a6365-5f24-449f-8e85-a0c745f5c13b)

**Comment by MarmadileManteater** [comment]
I recently made a change which might help with this problem. The `.db` files are now stored on the filesystem in [the latest nightly](https://github.com/MarmadileManteater/FreeTubeCordova/actions/runs/7768819416). This would not directly address these issues, but it may allow u to manually import/export your db files which can be found in `Android/data/io.freetubeapp.freetube.nightly/`.

I am also working on a new version of the app where I've entirely rewritten the process for importing/exporting db files. This version is currently incomplete. It may be buggy, but it genuinely might fix ur problem idk. The latest build for that can be found here:  https://github.com/MarmadileManteater/FreeTubeCordova/actions/runs/7790832175

**Comment by vdbhb59** [comment]
> I recently made a change which might help with this problem. The `.db` files are now stored on the filesystem in [the latest nightly](https://github.com/MarmadileManteater/FreeTubeCordova/actions/runs/7768819416). This would not directly address these issues, but it may allow u to manually import/export your db files which can be found in `Android/data/io.freetubeapp.freetube.nightly/`.
> 
> I am also working on a new version of the app where I've entirely rewritten the process for importing/exporting db files. This version is currently incomplete. It may be buggy, but it genuinely might fix ur problem idk. The latest build for that can be found here: https://github.com/MarmadileManteater/FreeTubeCordova/actions/runs/7790832175

Thanks, for the hard work. I will try out and give you the details as soon as I can. :)
